android/src/main/java/org/libsodium/rn/RCTSodiumModule.java

```diff
@@ -476,11 +476,32 @@ public void crypto_scalarmult_base(final String n, final Promise p) {
           p.resolve(Base64.encodeToString(q,Base64.NO_WRAP));
       }
     }
-    catch (Throwable t) {
+    catch (Throwable t) { 
       p.reject(ESODIUM,ERR_FAILURE,t);
     }
   }
 
+  @ReactMethod
+  public void crypto_scalarmult(final String n, final String q, final Promise p) {
```

Comment:
In native crypto_scalarmult n and p are the input parameters, while q is the output. Please correct parameter names of this method to reflect their purpose as in the original.